src/symtable/SymbolTableScope.ts

```diff
@@ -339,15 +345,16 @@ export class SymbolTableScope {
         classFlag: boolean
     ) {
         /* Update scope information for all symbols in this scope */
-        for (let [name, flags] of Object.entries(this.symbols)) {
+        for (const name in this.symbols) {
+            let flags = this.symbols[name];
             flags |= scopes[name] << SYMTAB_CONSTS.SCOPE_OFFSET;
             this.symbols[name] = flags;
         }
 
         /* Record not yet resolved free variables from children (if any) */
         const vFree = SYMTAB_CONSTS.FREE << SYMTAB_CONSTS.SCOPE_OFFSET;
 
-        for (const name in free) {
+        for (const name of free) {
```

Comment:
none of our tests apparently exercise this

Reply:
Yeah weird.